File: backend/src/main/java/com/atoria/backend/global/response/ApiResponse.java
package com.atoria.backend.global.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import org.springframework.http.HttpStatus;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record ApiResponse<T>(
        boolean success,
        int code,
        String message,
        T data,
        PageInfo pageInfo,
        String error
) {

    public static <T> ApiResponse<T> success(String message, T data) {
        return new ApiResponse<>(true, HttpStatus.OK.value(), message, data, null, null);
    }

    public static <T> ApiResponse<T> success(String message, T data, PageInfo pageInfo) {
        return new ApiResponse<>(true, HttpStatus.OK.value(), message, data, pageInfo, null);
    }

    public static <T> ApiResponse<T> created(String message, T data) {
        return new ApiResponse<>(true, HttpStatus.CREATED.value(), message, data, null, null);
    }

    public static <T> ApiResponse<T> accepted(String message, T data) {
        return new ApiResponse<>(true, HttpStatus.ACCEPTED.value(), message, data, null, null);
    }

    public static ApiResponse<Void> fail(HttpStatus status, String message, String error) {
        return new ApiResponse<>(false, status.value(), message, null, null, error);
    }
}

File: frontend/app/src/main/kotlin/com/ssafy/culture/data/remote/ApiResponse.kt
package com.ssafy.culture.data.remote

data class ApiResponse<T>(
    val success: Boolean,
    val code: Int,
    val message: String,
    val data: T?,
)

data class PageInfoDto(
    val page: Int,
    val size: Int,
    val totalElements: Int,
    val totalPages: Int,
)

data class PagedApiResponse<T>(
    val success: Boolean,
    val code: Int,
    val message: String,
    val data: T?,
    val pageInfo: PageInfoDto?,
)
